pipeline/inferrer/inference_manager/src/main/scala/uk/ac/wellcome/platform/inference_manager/adapters/InferrerAdapter.scala

```diff
@@ -0,0 +1,48 @@
+package uk.ac.wellcome.platform.inference_manager.adapters
+
+import akka.http.scaladsl.model.{HttpRequest, HttpResponse, StatusCodes}
+import akka.http.scaladsl.unmarshalling.Unmarshal
+import akka.stream.Materializer
+import de.heikoseeberger.akkahttpcirce.FailFastCirceSupport._
+import grizzled.slf4j.Logging
+import io.circe.Decoder
+import uk.ac.wellcome.models.work.internal.InferredData
+import uk.ac.wellcome.platform.inference_manager.models.DownloadedImage
+
+import scala.concurrent.Future
+
+/*
+ * An InferrerAdapter is specific to the inferrer and the data that is being augmented.
+ * Implementors must provide:
+ * - The type of the inferrer response
+ * - A Decoder for that response
+ * - A function to create an HTTP request from the input data type
+ * - A function to augment input data with a response, returning the output data type
+ *
+ * Additionally, the trait provides the logic for handling different HTTP response statuses
+ */
+trait InferrerResponse
```

Comment:
I don't see the need of having this trait personally, as it's not a sealed trait allowing matching or an interface which subtypes have to fulfill. Would be just as clear with `type Response` below imo